L1Trigger/L1THGCal/plugins/backend/HGCalTowerProcessor.cc

```diff
@@ -7,23 +7,64 @@
 #include "Geometry/Records/interface/CaloGeometryRecord.h"
 #include "L1Trigger/L1THGCal/interface/HGCalTriggerGeometryBase.h"
 #include "L1Trigger/L1THGCal/interface/HGCalTriggerBackendAlgorithmBase.h"
+#include "L1Trigger/L1THGCal/interface/concentrator/HGCalConcentratorTrigSumImpl.h"
+#include "L1Trigger/L1THGCal/interface/backend/HGCalTowerMap2DImpl.h"
 #include "L1Trigger/L1THGCal/interface/backend/HGCalTowerMap3DImpl.h"
 
 class HGCalTowerProcessor : public HGCalTowerProcessorBase {
 public:
   HGCalTowerProcessor(const edm::ParameterSet& conf) : HGCalTowerProcessorBase(conf) {
+    trigSumImpl_ = std::make_unique<HGCalConcentratorTrigSumImpl>(conf);
+    towermap2D_ = std::make_unique<HGCalTowerMap2DImpl>(conf.getParameterSet("towermap_parameters"));
     towermap3D_ = std::make_unique<HGCalTowerMap3DImpl>();
   }
 
-  void run(const edm::Handle<l1t::HGCalTowerMapBxCollection>& collHandle,
+  void eventSetup(const edm::EventSetup& es) override {
+    trigSumImpl_->eventSetup(es);
+    towermap2D_->eventSetup(es);
+  }
+
+  void run(const std::pair<edm::Handle<l1t::HGCalTowerMapBxCollection>, edm::Handle<l1t::HGCalClusterBxCollection>>&
+               collHandle,
            l1t::HGCalTowerBxCollection& collTowers,
            const edm::EventSetup& es) override {
     es.get<CaloGeometryRecord>().get("", triggerGeometry_);
 
+    auto& towerMapCollHandle = collHandle.first;
+    auto& unclTCsCollHandle = collHandle.second;
+
     /* create a persistent vector of pointers to the towerMaps */
     std::vector<edm::Ptr<l1t::HGCalTowerMap>> towerMapsPtrs;
-    for (unsigned i = 0; i < collHandle->size(); ++i) {
-      edm::Ptr<l1t::HGCalTowerMap> ptr(collHandle, i);
+    for (unsigned i = 0; i < towerMapCollHandle->size(); ++i) {
+      edm::Ptr<l1t::HGCalTowerMap> ptr(towerMapCollHandle, i);
+      towerMapsPtrs.push_back(ptr);
```

Comment:
Can do directly:
```cpp
towerMapsPtrs.emplace_back(towerMapCollHandle, i);
```
Which would avoid the copy (although in principle it should be rather cheap for a `Ptr`)